src/devices/machine/smc91c9x.cpp

```diff
@@ -181,25 +214,210 @@ void smc91c9x_device::device_reset()
 	m_reg[EREG_ERCV]         = 0x331f;   m_regmask[EREG_ERCV]         = 0x009f;
 
 	update_ethernet_irq();
-	m_tx_timer->adjust(attotime::never);
 }
 
-
 DEFINE_DEVICE_TYPE(SMC91C94, smc91c94_device, "smc91c94", "SMC91C94 Ethernet Controller")
 
 smc91c94_device::smc91c94_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
 	: smc91c9x_device(mconfig, SMC91C94, tag, owner, clock)
 {
 }
 
-
 DEFINE_DEVICE_TYPE(SMC91C96, smc91c96_device, "smc91c96", "SMC91C96 Ethernet Controller")
 
 smc91c96_device::smc91c96_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
 	: smc91c9x_device(mconfig, SMC91C96, tag, owner, clock)
 {
 }
 
+void smc91c9x_device::clear_tx_fifo()
+{
+	tx_fifo_in = 0;
+	tx_fifo_out = 0;
+	memset(m_tx, 0, sizeof(m_tx));
```

Comment:
Please use `std::fill(std::begin(x), std::end(x), val)` for filling arrays - it will cause an error if the array is replaced with a pointer or something rather than just silently doing the wrong thing.

Reply:
Done in the new PR.